src/device/selection.ts

```diff
@@ -231,12 +231,6 @@ export const set = (
   const range = document.createRange()
   const sel = window.getSelection() || new Selection()
 
-  // bail if already selected
```

Comment:
## Reason for removing this particular logic
- So this scenario becomes different when we are trying to paste something on the same node(focusNode). With this logic it will never allow the caret position to be set as we will be pasting content on the same node. Hence I don't think this logic is necessary. I am not fully sure at the first place why this logic exists. If you have any objections regarding this please enlighten me!

Reply:
@karunkop Does this still need to be removed with the new approach?